src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groupedItems: GroupsMap<T>, item: T) => {
+    const keyValue = item[key] as string;
```

Comment:
The line `const keyValue = item[key] as string;` assumes that the key value is always a string. This might not be the case if the key is of a different type (e.g., number or symbol). Consider using `String(item[key])` to ensure the key is treated as a string for indexing purposes.